src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsuserpreferences/service/PreferencesService.kt

```diff
@@ -26,9 +26,10 @@ class PreferencesService {
     val preferences = preferencesDto.items
       .map { item -> Preference(userId, preferenceName, item) }
 
-    preferenceRepository.deleteAll(preferenceRepository.findByHmppsUserIdAndName(userId, preferenceName))
+    val currentPreferences = preferenceRepository.findByHmppsUserIdAndName(userId, preferenceName)
+    preferenceRepository.deleteAll(currentPreferences)
```

Comment:
Here `currentPreferences` refers to the preferences before they've been updated whereas in the TelemetryService the same thing is referred to as `previousPreferenceValues`. It would be better to be consistent, I think `previous` is a bit clearer about the intent.

Reply:
Addressed, thanks